gmso/core/topology.py

```diff
@@ -965,6 +966,171 @@ def get_index(self, member):
 
         return index
 
+    def to_datatables(self, parameter="sites", labels=None, unyts_bool=True):
+        """Return a pandas dataframe object for the sites in a topology
+
+        Parameters
+        ----------
+        parameter : a string
+            A string determining what aspects of the gmso topology will be reported. Options are: 'sites','bonds','angles'
+        and 'dihedrals'. Defaults to 'sites'.
```

Comment:
are we planning to add improper later?

Reply:
Improper support is now added, I probably have to write unit testing to make sure that functions alright. Need a bit more testing on that end because I'm less familiar with systems that have impropers.